pyaerocom/aeroval/setupclasses.py

```diff
@@ -1,10 +1,15 @@
-from __future__ import annotations
-
 import logging
 import os
+import sys
 from getpass import getuser
 from pathlib import Path
-from typing import Literal, Self
+
+if sys.version_info < (3, 10):
+    from typing_extensions import Self
+else:
+    from typing import Self
+
+from typing import Literal
```

Comment:
`Self` was added to the `typing` module on Python 3.11, see [PEP 673](https://peps.python.org/pep-0673/), so you want the if written as follows

``` python
from typing import Literal

if sys.version_info >= (3, 11):
    from typing import Self
else:
    from typing_extensions import Self
```

also by moving the `Literal` import up, the conditional import sits between imports from the standard library and from 3rd party modules.

Reply:
Weird, this is an outdated version. In the newer version of the code check to see if `sys.version_info < (3, 11)`, because of the reason you mention. I agree with you styling of the import though, so I will change it to a >= 3.11 statement so it is organized this way.